single-pool/js/packages/client/src/quarantine.ts

```diff
@@ -0,0 +1,264 @@
+    ];
+
+    return {
+      data,
+      accounts,
+      programAddress: ATOKEN_PROGRAM_ID,
+    };
+  }
+}
+
+export enum StakeAuthorizationType {
+  Staker,
+  Withdrawer,
+}
+
+export class StakeInstruction {
+  // idc about doing it right unless this goes in a lib
+  static initialize(params: {
+    stakeAccount: Base58EncodedAddress;
+    staker: Base58EncodedAddress;
+    withdrawer: Base58EncodedAddress;
+  }) {
+    const { serialize } = getBase58EncodedAddressCodec();
+    const data = new Uint8Array([
+      ...u32(0),
+      ...serialize(params.staker),
+      ...serialize(params.withdrawer),
+      ...Array(48).fill(0),
+    ]);
+
+    const accounts = [
+      { address: params.stakeAccount, role: AccountRole.WRITABLE },
+      { address: SYSVAR_RENT_ID, role: AccountRole.READONLY },
+    ];
+
+    return {
+      data,
+      accounts,
+      programAddress: STAKE_PROGRAM_ID,
+    };
+  }
+
+  static authorize(params: {
+    stakeAccount: Base58EncodedAddress;
+    authorized: Base58EncodedAddress;
+    newAuthorized: Base58EncodedAddress;
+    authorizationType: StakeAuthorizationType;
+    custodian?: Base58EncodedAddress;
+  }) {
+    const { serialize } = getBase58EncodedAddressCodec();
+    const data = new Uint8Array([
+      ...u32(1),
+      ...serialize(params.newAuthorized),
+      ...u32(params.authorizationType),
+    ]);
+
+    const accounts = [
+      { address: params.stakeAccount, role: AccountRole.WRITABLE },
+      { address: SYSVAR_CLOCK_ID, role: AccountRole.READONLY },
+      { address: params.authorized, role: AccountRole.READONLY_SIGNER },
+    ];
+    if (params.custodian) {
+      accounts.push({ address: params.custodian, role: AccountRole.READONLY });
+    }
+
+    return {
+      data,
+      accounts,
+      programAddress: STAKE_PROGRAM_ID,
+    };
+  }
+
+  static delegate(params: {
+    stakeAccount: Base58EncodedAddress;
+    authorized: Base58EncodedAddress;
+    voteAccount: Base58EncodedAddress;
+  }) {
+    const data = new Uint8Array(u32(2));
+
+    const accounts = [
+      { address: params.stakeAccount, role: AccountRole.WRITABLE },
+      { address: params.voteAccount, role: AccountRole.READONLY },
+      { address: SYSVAR_CLOCK_ID, role: AccountRole.READONLY },
+      { address: SYSVAR_STAKE_HISTORY_ID, role: AccountRole.READONLY },
+      { address: STAKE_CONFIG_ID, role: AccountRole.READONLY },
+      { address: params.authorized, role: AccountRole.READONLY_SIGNER },
+    ];
+
+    return {
+      data,
+      accounts,
+      programAddress: STAKE_PROGRAM_ID,
+    };
+  }
+}
+
+export async function getAssociatedTokenAddress(
+  mint: Base58EncodedAddress,
+  owner: Base58EncodedAddress,
+) {
+  const { serialize } = getBase58EncodedAddressCodec();
+  const { pda } = await getProgramDerivedAddress({
+    programAddress: ATOKEN_PROGRAM_ID,
+    seeds: [serialize(owner), serialize(TOKEN_PROGRAM_ID), serialize(mint)],
+  });
+
+  return pda;
+}
```

Comment:
This one you _could_ pull from `@solana/spl-token`, but it hasn't gotten the experimental treatment yet, so you'd have to use the `@solana/compat` library @mcintyre94 just started writing, which basically does what some of this file is doing. Not sure if you want those two dependencies though

Reply:
i had tried using spl-token but couldnt because its not possible to have both versions of web3.js in one project

Reply:
ohh yeah duh lol